test/e2e/funcs/feature.go

```diff
@@ -133,6 +133,100 @@ func DeploymentBecomesAvailableWithin(d time.Duration, namespace, name string) f
 	}
 }
 
+// DeploymentPodIsRunningMustNotChangeWithin fails a test if the supplied Deployment does
+// not have a running Pod that stays running for the supplied duration.
+func DeploymentPodIsRunningMustNotChangeWithin(d time.Duration, namespace, name string) features.Func {
+	return func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		t.Helper()
+
+		dp := &appsv1.Deployment{ObjectMeta: metav1.ObjectMeta{Namespace: namespace, Name: name}}
+		t.Logf("Ensuring deployment %s/%s has running pod that does not change within %s", dp.GetNamespace(), dp.GetName(), d.String())
+		start := time.Now()
+
+		pod, err := podForDeployment(ctx, t, c, dp)
+		if err != nil {
+			t.Errorf("Failed to get pod for deployment %s/%s: %s", dp.GetNamespace(), dp.GetName(), err)
+			return ctx
+		}
+
+		// first wait for pod to be running
+		if err := wait.For(conditions.New(c.Client().Resources()).PodConditionMatch(pod, corev1.PodReady, corev1.ConditionTrue), wait.WithTimeout(d), wait.WithInterval(DefaultPollInterval)); err != nil {
+			t.Errorf("Deployment %s/%s never got a running pod after %s", dp.GetNamespace(), dp.GetName(), since(start))
+			return ctx
+		}
+
+		// now wait to make sure the pod stays running (does not change)
+		start = time.Now()
+		if err := wait.For(conditions.New(c.Client().Resources()).PodConditionMatch(pod, corev1.PodReady, corev1.ConditionFalse), wait.WithTimeout(d), wait.WithInterval(DefaultPollInterval)); err != nil {
+			if deadlineExceed(err) {
+				t.Logf("Deployment %s/%s had running pod that did not change after %s", dp.GetNamespace(), dp.GetName(), since(start))
+			} else {
+				t.Errorf("Error while observing pod for deployment %s/%s", dp.GetNamespace(), dp.GetName())
+			}
```

Comment:
```suggestion
			if deadlineExceed(err) {
				t.Logf("Deployment %s/%s had running pod that did not change after %s", dp.GetNamespace(), dp.GetName(), since(start))
				return ctx
			t.Errorf("Error while observing pod for deployment %s/%s", dp.GetNamespace(), dp.GetName())
```

Style nit: I prefer to return early to reduce nesting where possible. This captures that the "Error while observing..." is kinda the default/catchall case for the block it's in.

Reply:
yep that's a good pattern and the intent of this block is preserved with an earlier return.

this same type of check was in 3 places, so I updated all 3 in https://github.com/crossplane/crossplane/pull/6182/commits/93f96e6eff93f256333256a0c80b50417fcafadb.